packages/astro/src/actions/utils.ts

```diff
@@ -25,3 +27,46 @@ export function createCallAction(context: ActionAPIContext): APIContext['callAct
 		return action(input) as any;
 	};
 }
+
+let didInitLexer = false;
+
+export async function usesActions(fs: typeof fsMod, srcDir: URL) {
```

Comment:
Can we add a comment that explains what this function does? The logic is really weird and I can't understand it

Reply:
Good call. I renamed this to `isActionsFilePresent` and added an in-line comment on why we use lexer.